src/views/settings/SettingsSubjects.tsx

```diff
@@ -315,39 +318,104 @@ const SettingsSubjects: Screen<"SettingsSubjects"> = ({ navigation }) => {
+                                    borderColor: colors.background,
+                                    borderWidth: 3,
+                                  }}
+                                  entering={ZoomIn.springify().mass(1).damping(20).stiffness(300)}
+                                  exiting={ZoomOut.springify().mass(1).damping(20).stiffness(300)}
+                                />
+                              )}
+                            </View>
+                          </TouchableOpacity>
+                        );
+                      }}
+                    />
+                    <MemoizedNativeList>
+                      <MemoizedNativeItem>
+                        <MemoizedNativeText variant="subtitle" numberOfLines={1}>
+                          Ou écrire la couleur (6 caractères)
+                        </MemoizedNativeText>
+                        <View style={{ flexDirection: "row", alignItems: "center", marginTop: 10 }}>
                           <View
                             style={{
-                              width: 32,
-                              height: 32,
-                              borderRadius: 80,
-                              backgroundColor: item,
-                              marginHorizontal: 5,
-                              alignItems: "center",
-                              justifyContent: "center",
+                              width: 26,
+                              height: 26,
+                              backgroundColor: /^#[0-9A-F]{6}$/i.test(customColor) ? customColor : colors.text,
+                              borderWidth: 1,
+                              borderColor: colors.border,
+                              marginRight: 8,
+                              borderRadius: 80
+                            }}
+                          />
+                          <TextInput
+                            style={{
+                              fontFamily: "medium",
+                              fontSize: 16,
+                              color: colors.text,
+                              flex: 1,
+                            }}
+                            value={customColor !== "" ? customColor : "#"}
+                            onChangeText={(text) => {
+                              if (!text.startsWith("#")) {
+                                text = "#" + text;
+                              }
+                              text = text.replaceAll(/[G-Z]/g, "");
+                              text = text.slice(0, 7).toUpperCase();
+                              setCustomColor(text);
+                            }}
+                          />
+                          <TouchableOpacity
+                            onPress={() => {
+                              if (/^#[0-9A-F]{6}$/i.test(customColor)) {
+                                handleSubjectColorChange(selectedSubject[0], customColor);
+                                setCustomColor("");
```

Comment:
non faut pas. Mais on va rendre ça encore plus intéractif !
```suggestion
                                setSelectedSubject((prev) => {
                                  if (prev) {
                                    prev[1].color = customColor;
                                  }
                                  return prev;
                                });
```